File: android/src/main/kotlin/com/zpdl_studio/zpdl_studio_media_plugin/data/PluginImageInfo.kt
package com.zpdl_studio.zpdl_studio_media_plugin.data

data class PluginImageInfo(
        val id: Long,
        val path: String,
        val displayName: String,
        val mimeType: String,
        val orientation: Int,
        val width: Int,
        val height: Int,
        val modifyTimeMs: Long
) : PluginToMap {
    override fun pluginToMap(): Map<String, *> = hashMapOf(
            "id" to id,
            "path" to path,
            "displayName" to displayName,
            "mimeType" to mimeType,
            "orientation" to orientation,
            "width" to width,
            "height" to height,
            "modifyTimeMs" to modifyTimeMs,
    )
}

File: ios/Classes/data/PluginImageInfo.swift
//
//  PluginImageInfo.swift
//  zpdl_studio_media_plugin
//
//  Created by 김경환 on 2020/10/15.
//

struct PluginImageInfo: PluginToMap {
    let id: String
    let fullPath: String
    let mimeType: String
    let orientation: Int

    func pluginToMap() -> [String : Any] {
        return [
            "fullPath": fullPath,
            "displayName": id,
            "mimeType": mimeType,
            "orientation": orientation,
        ]
    }
}
